### DISCUSSION TITLE
Every time I try to save my mod settings I get an error saying it's being used by another process, and whenever I close the launcher it gives me this error message.

### DISCUSSION THREAD

**Discussion by Sylthana** [X2CommunityCore/xcom2-launcher#313] [Help] [answered]
Every time I try to save my mod settings I get an error saying it's being used by another process, and whenever I close the launcher it gives me this error message. 

`AML version: 1.5.0
Sentry GUID: efd39e90-d0d7-44a4-8c43-d07e987add18
Message: The process cannot access the file because it is being used by another process.

Stack: 
    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalReplace(String sourceFileName, String destinationFileName, String destinationBackupFileName, Boolean ignoreMetadataErrors)
   at XCOM2Launcher.Program.Main() in f:\AML\xcom2-launcher\xcom2-launcher\Program.cs:line 145`

although sometimes it gives this message instead, seemingly at random. 

`AML version: 1.5.0
Sentry GUID: efd39e90-d0d7-44a4-8c43-d07e987add18
Message: The process cannot access the file because it is being used by another process.

Stack: 
    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalReplace(String sourceFileName, String destinationFileName, String destinationBackupFileName, Boolean ignoreMetadataErrors)
   at XCOM2Launcher.Program.InitAppSettings() in f:\AML\xcom2-launcher\xcom2-launcher\Program.cs:line 297
   at XCOM2Launcher.Program.Main() in f:\AML\xcom2-launcher\xcom2-launcher\Program.cs:line 83`

I've tried restarting the launcher, restarting steam, restarting my computer, reinstalling the mod launcher, updating the launcher. 

I don't check github too often so if I don't respond, I'm @sylthana on the xcom 2 modding discord, so it'd probably be faster to ping me.

**Comment by RevZero** [comment, marked as answer]
Hi. 
What is your path / installation folder for AML? AML requires a folder with write access. The safest option is, to create a folder on your Desktop. **Avoid system folders** like _c:\Program Files\_ or _c:\Windows_.
It is also possible, that some Antivirus blocks write access. If you have one running, try to temporarily disable it and see if that helps.

**Reply by Sylthana** [reply]
I've got the folder on my desktop. As for anti-virus, I'll get back to you on that one. My computer's being finicky right now. Although I know McAfee likes to mess with me sometimes.

**Reply by Sylthana** [reply]
Apparently I uninstalled mcafee a while back because of how much trouble it was causing and how little it was actually helping. That and windows defender doesn't tend to do anything aside from warn me that a file *might* be dangerous, after clicking run anyway it usually backs off. It did this when I ran the launcher for the first time, although it also worked fine the first time I used it.

**Reply by RevZero** [reply]
If you have the folder on your desktop and get "The process cannot access the file because it is being used by another process.", its very likely related to some security app blocking access. You could try extracting AML to another folder and see what happens. Maybe WinDefender asks again. You could also set the whole AML folder as an exception https://support.microsoft.com/en-us/windows/add-an-exclusion-to-windows-security-811816c0-4dfd-af4a-47e4-c301afe13b26

**Reply by Sylthana** [reply]
I got home and started up my computer to try that. I did none of it, and apparently AML started working all on it's own.

**Reply by RevZero** [reply]
It would have been nice to find out the cause, but at least it works 🤷🏻‍♂️
